File: TPs/TP01/Q06-ok/Is.java
/**
Crie um método iterativo que recebe uma string e
*   retorna true se a mesma é composta somente por vogais.
Crie outro método iterativo que recebe uma string e
*   retorna true se a mesma é composta somente por consoantes.
Crie um terceiro método iterativo que recebe uma string e
retorna true se a mesma corresponde a um número inteiro.
Crie um quarto método iterativo que recebe uma string e
*   retorna true se a mesma corresponde a um número real.
Na saı́da padrão, para cada linha de entrada, escreva outra de saı́da da seguinte forma
*   X1 X2 X3 X4 onde cada Xi é um booleano indicando se a é entrada é:
*   composta somente por vogais (X1);
*   composta somente somente por consoantes (X2);
*   um número inteiro (X3);
*   um número real (X4).
*   Se Xi for verdadeiro, seu valor será SIM, caso contrário, NÃO.
*/
import java.lang.Math;
import java.text.NumberFormat;
class Is {
    /**
    Encontrar fim
    @param  String a testar
    @return true -> fim     encontrado
    *       false-> fim nao encontrado
    */
    public static boolean isFim (String s){
        return (s.length()==3 && s.charAt(0)=='F'&& s.charAt(1)=='I'&& s.charAt(2)=='M');
    }
    /**
    X1      -> Only Vowels
    @param  -> String a ser testada
    @return ->  true    -> se nao encontrada consoante
    *           false   -> se     encontrada consoante
    */
    public static void x1(String s){
        //array base
        char[] vowels = {'a','e','i','o','u','A','E','I','O','U'};
        //inicio cod. pesquisa
        boolean flag = false;
        //pesquise sequencial (string recebida)
        for(int i=0; i<s.length(); i++){
            //flag de resultado de pesq seq
            flag = false;
                //print teste
                //MyIO.println("s->"+s.charAt(i));
            //pesquisa sequencial (vogais)
            for(int j=0; j<10; j++){
                    //print teste
                    //MyIO.println("vowels->"+vowels[j]);
                //testa se char da string == vogal
                if(s.charAt(i)==vowels[j]){
                    flag = true;
                        //print teste
                        //MyIO.println("found->"+vowels[j]);
                    //finaliza a pesquisa
                    j=10;
                }
            }
            //se encontrado consoante nessa letra parar
            if (flag==false){
                //return flag;
                flag = false;
                break;
            }
        }
        //primeiro print n tem espaco
        if(flag == true)
            {MyIO.print("SIM");}
        else
            {MyIO.print("NAO");}
    }
    /**
    X2  -> Only Consonants
    @param  -> String a ser testada
    @return ->  true    -> se nao encontrada vogal
    *           false   -> se     encontrada vogal
    */
    public static boolean x2(String s){
        char[] vowels = {'a','e','i','o','u','A','E','I','O','U'};
        boolean flag = false;
        //pesquise sequencial (string recebida)
        for(int i=0; i<s.length(); i++){
            flag = true;
            //testar se numero
            if(s.charAt(i)<65){
                return flag = false;
            }
            //pesquisa sequencial (vogais)
            for(int j=0; j<10; j++){
                //testa se char da string == vogal
                if(s.charAt(i)==vowels[j]){
                    return flag = false;
                }
            }
        }
        return flag;
    }
    /**
    X3  -> Only Integers
    @param  -> String a ser testada
    @return ->  true    -> se encontrado inteiro
    *           false   -> se nao encontrado inteiro
    */
    public static boolean x3(String s){
        boolean flag = false;
        //
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)>=48 && s.charAt(i)<=57){
                try{
                    //alocacao para uso do try
                    int temp = Integer.parseInt(s);
                    //System.out.println("\n"+temp);
                    flag = true;
                    temp = 0;
                } catch (NumberFormatException ex){
                    return flag;
                }
            }
        }
        return flag;
    }
    /**
    X4  -> Only Doubles
    */
    public static boolean x4(String s){
        boolean flag = false;
        //
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)>=48 && s.charAt(i)<=57){
                // tentar como int
                try{
                    //inteiro tambem eh real
                    int numero = Integer.parseInt(s);
                    //testar se realmente int
                    if(numero%1==0){
                        return flag = true;
                    }
                    //numero nao identificado
                    else{
                        return flag = false;
                    }
                // exception -> nao eh int
                } catch (NumberFormatException ex){
                    //tentar como double 0.0
                    try{
                        Double numero = Double.parseDouble(s);
                        numero++;
                        //System.out.println("\n"+numero);
                        return flag = true;
                    }
                    //exception -> double 0,0
                    catch(NumberFormatException ex1){
                        /** tentativa de identificar double 0,0
                        try{
                            NumberFormat format = NumberFormat.getInstance(Locale.FRANCE);
                            Number number = format.parse(s);
                            Double numero = number.doubleValue();
                            //testar se double
                            if(numero%1!=0){
                                return flag = true;
                            }
                            //erro, numero n identificado
                            else{
                                return flag = false;
                            }
                        }
                        //nao identificado
                        catch (NumberFormatException ex3){
                            return flag = false;
                        }*/
                        return flag = false;
                    }
                }
            }
        }
        return flag;
    }
    /**
    */
    public static void Print(boolean resp){
        if(resp == true)
            MyIO.print(" SIM");
        else
            MyIO.print(" NAO");
    }
    /**
    Main
    */
    public static void main (String[] args){
        //MyIO.setCharset("UTF-8");
        //Variaveis de entrada
        String[] entrada = new String[1000];
        int numEntrada=0;
        //Leitura da entrada padrao
        // while not 'FIM'
        do{
            entrada[numEntrada]=MyIO.readLine();
            //If para impedir e entrada da palavra fim
            if (isFim(entrada[numEntrada])==false){
                //Impressora
                //se vogal
                //boolean temp1=x1(entrada[numEntrada]);
                //Print(temp1);
                x1(entrada[numEntrada]);
                //se consoante
                boolean temp2=x2(entrada[numEntrada]);
                Print(temp2);
                //se int
                boolean temp3=x3(entrada[numEntrada]);
                Print(temp3);
                //se double
                boolean temp4=x4(entrada[numEntrada]);
                Print(temp4);
                //proxima linha
                MyIO.println("");
            }
        } while (isFim(entrada[numEntrada++])==false);
        numEntrada--;   //Desconsidera a linha da palavra FIM
    }
}

File: TPs/TP01/Q07-ok/Is.c
#include <stdio.h>
#include <string.h> 
#include <stdbool.h>
#include <math.h>

#define NUMLINHA  1000
#define TAMPALAVRA 100

/**
Encontrar fim
@param  String a testar
@return true -> fim     encontrado
*       false-> fim nao encontrado
*/
bool isFim (char s[]){
    return (strlen(s)>=3 && s[0]=='F' && s[1]=='I' && s[2]=='M');
}
/**
X1      -> Only Vowels
@param  -> String a ser testada
@return ->  true    -> se nao encontrada consoante
*           false   -> se     encontrada consoante
*/
void x1(char s[]){
    char vowels[] = {'a','e','i','o','u','A','E','I','O','U'};
    //inicio cod. pesquisa
    bool flag = false;
    for(int i=0; i<strlen(s); i++){
        flag = false;
        for(int j=0; j<10; j++){
            if(s[i]==vowels[j]){
                j=10;
                flag=true;
            }
        }
        //se encontrado consoante nessa letra parar
        if (flag==false){
            flag = false;
            printf("NAO");
            break;
        }
    }
    //primeiro print n tem espaco
    if(flag == true)
        {printf("SIM");}
}
/**
X2  -> Only Consonants
@param  -> String a ser testada
@return ->  true    -> se nao encontrada vogal
*           false   -> se     encontrada vogal
*/
bool x2(char s[]){
    char vowels[] = {'a','e','i','o','u','A','E','I','O','U'};
    //inicio cod. pesquisa
    int last = strlen(s);
    bool flag = false;
    for(int i=0; i<last; i++){
        flag = true;
        if(s[i]>65){
            for(int j=0; j<10; j++){
                if(s[i]==vowels[j]){
                    return flag = false;
                }
            }
        }
        else{
            return flag = false;
        }
    }
    return flag;
}
/**
X3  -> Only Integers
@param  -> String a ser testada
@return ->  true    -> se encontrado inteiro
*           false   -> se nao encontrado inteiro
*/
bool x3(char s[]){
    bool flag = false;
    int last = strlen(s);
    //inicio cod. pesquisa
    for(int i=0; i<last; i++){
        //48 -> '0'
        //57 -> '9'
        if(s[i]>=48 && s[i]<=57){
            //46 -> '.'
            //44 -> ','
            if(s[i]!=46 || s[i]!=44){
                flag = true;
            }
            else{
                return flag = false;
            }
        }
        else{
            return flag = false;
        }
    }
    return flag;
}
/**
X4  -> Only Doubles
*/
bool x4(char s[]){
    bool flag = false;
    int last = strlen(s);
    //+1 ponto ou virgula = false
    int tempPonto=0;
    //inicio cod. pesquisa
    for(int i=0; i<last; i++){
        //48 -> '0'
        //57 -> '9'
        //46 -> '.'
        //44 -> ','
        if(s[i]>=48 && s[i]<=57 || s[i]==46 || s[i]==44){
            //testa quantos pontos
            if(s[i]==46 || s[i]==44){
                tempPonto++;
                flag = true;
                if(tempPonto>1){
                    return flag = false;
                }
            }
        }
        else{
            return flag = false;
        }
    }
    return flag;
}
/**
*/
void Print(bool resp){
    if(resp == true)
        printf(" SIM");
    else
        printf(" NAO");
}
/**
Main
*/
int main (){
    char entrada[NUMLINHA][TAMPALAVRA];
    int numEntrada=0;
    //Leitura da entrada padrao
    do{
        scanf("%[^\n]%*c", entrada[numEntrada]);
        if (isFim(entrada[numEntrada])==false){
            //se vogal
            x1(entrada[numEntrada]);
            //se consoante
            bool temp2=x2(entrada[numEntrada]);
            Print(temp2);
            //se int
            bool temp3=x3(entrada[numEntrada]);
            Print(temp3);
            //se double
            bool temp4=x4(entrada[numEntrada]);
            Print(temp4);
            //proxima linha
            printf("\n");
        }
    } while (isFim(entrada[numEntrada++])==false);
    numEntrada--;   //Desconsidera a linha da palavra FIM
    //
    return 0;
}